engines/registry/app/controllers/registry/registry_controller.rb

```diff
@@ -19,12 +19,23 @@ def authorize
     # Returns a Distribution Registry HTTP API V2 - compatible repository catalog as defined in
     # https://distribution.github.io/distribution/spec/api/#listing-repositories
     def catalog
+      raise 'Could not find system with current credentials' if @client&.account.nil?
```

Comment:
Can we classify this as a specific exception, instead of handling with the all-encompassing StandardError? I'm worried your handler is going to mask other exceptions.

Reply:
The conditional could be simplified as `unless @client&.account`, or you could go for readability/consistency with `unless @client && @client.account`... but the `.nil?` seems superfluous.

Reply:
Resolved in 3b832353.